### ISSUE TITLE
Option -save_model does not work in transformer mode

### ISSUE THREAD

**Issue by tanaka-jp** [OpenNMT/OpenNMT-py#848]
python train.py -accum_count 4 -adam_beta2 0.998 -batch_size 2048 -batch_type tokens -copy_attn -data data_do/baseline1 -decay_method noam -decoder_type transformer -dropout 0.2 -encoder_type transformer -gpuid 0 -label_smoothing 0.1 -layers 4 -learning_rate 2 -max_generator_batches 2 -max_grad_norm 0 -normalization tokens -optim adam -param_init 0 -param_init_glorot -position_encoding -rnn_size 512 -save_model mymodel -share_embeddings -train_steps 100 -warmup_steps 8000 -word_vec_size 512

My version of OpenNMT-py is 0.2.
No error occurred. But the model is not saved

[2018-07-24 13:12:53,590 INFO] encoder: 38212608
[2018-07-24 13:12:53,590 INFO] encoder: 38212608
[2018-07-24 13:12:53,591 INFO] decoder: 43256149
[2018-07-24 13:12:53,591 INFO] decoder: 43256149
[2018-07-24 13:12:53,591 INFO] * number of parameters: 81468757
[2018-07-24 13:12:53,591 INFO] * number of parameters: 81468757
[2018-07-24 13:12:53,594 INFO] Start training...
[2018-07-24 13:12:53,594 INFO] Start training...
[2018-07-24 13:13:00,138 INFO] Loading train dataset from data_do\baseline1.train.1.pt, number of ex
amples: 44083
[2018-07-24 13:13:00,138 INFO] Loading train dataset from data_do\baseline1.train.1.pt, number of ex
amples: 44083
[2018-07-24 13:16:21,693 INFO] Step 50/  100; acc:   5.05; ppl: 2389.35; xent: 7.78; lr: 0.00001; 10
17/2028 tok/s;    202 sec
[2018-07-24 13:16:21,693 INFO] Step 50/  100; acc:   5.05; ppl: 2389.35; xent: 7.78; lr: 0.00001; 10
17/2028 tok/s;    202 sec
[2018-07-24 13:19:44,512 INFO] Step 100/  100; acc:   5.41; ppl: 2418.75; xent: 7.79; lr: 0.00001; 8
93/1890 tok/s;    404 sec
[2018-07-24 13:19:44,512 INFO] Step 100/  100; acc:   5.41; ppl: 2418.75; xent: 7.79; lr: 0.00001; 8
93/1890 tok/s;    404 sec
[2018-07-24 13:19:50,997 INFO] Loading train dataset from data_do\baseline1.train.1.pt, number of ex
amples: 44083
[2018-07-24 13:19:50,997 INFO] Loading train dataset from data_do\baseline1.train.1.pt, number of ex
amples: 44083

**Comment by tanaka-jp** [comment]
This problem is also happening in rnn model.

**Comment by vince62s** [comment]
read this: https://github.com/OpenNMT/OpenNMT-py/blob/master/onmt/opts.py#L231-L240
if you keep save every 5000 steps and train only 100 steps, it won't save.

**Comment by tanaka-jp** [comment]
It worked. Thank you very much
